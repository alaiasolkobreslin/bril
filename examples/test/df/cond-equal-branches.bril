# CMD: bril2json < {filename} | python3 ../../l4.py {args}
#
@main(cond: bool) {
  a: int = const 47;
  b: int = const 42;
  br cond .left .right;
.left:
  b: int = const 1;
  c: int = const 10;
  jmp .end;
.right:
  a: int = const 47;
  b: int = const 100;
  c: int = const 10;
  jmp .end;
.end:
  d: int = sub a c;
  print d;
}
