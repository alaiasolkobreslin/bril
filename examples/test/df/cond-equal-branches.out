b0:
	in:  {}
	out:  a: 47, b: 42
left:
	in:  a: 47, b: 42
	out:  a: 47, b: 1, c: 10
right:
	in:  a: 47, b: 42
	out:  a: 47, b: 100, c: 10
end:
	in:  a: 47, b: ?, c: 10
	out:  a: 47, b: ?, c: 10, d: ?
